validator/validator.go

```diff
@@ -0,0 +1,197 @@
+	var missingColumns []string
+	for _, mandatoryField := range mandatoryFields {
+		if _, ok := columns[mandatoryField]; !ok {
+			missingColumns = append(missingColumns, mandatoryField)
+		}
+		delete(columns, mandatoryField)
+	}
+
+	// If all the mandatory fields have been found then we don't need to return an error. However,
+	// if there is at least one mandatory field missing in the schema then this queue is invalid.
+	// TODO: Add some more logic to maybe indicate which field is the one that need to be included
+	if len(missingColumns) > 1 {
+		return errors.Errorf("some PGQ columns are missing: %v", missingColumns)
+	}
+
+	// TODO log extra columns in queue table or ignore them?
+	// extraColumns := make([]string, 0, len(columns))
+	// for k := range columns {
+	//	extraColumns = append(extraColumns, k)
+	// }
+	// _ = extraColumns
+
+	return nil
+}
+
+// Validate if requiered indexes exist
+func ValidateIndexes(db *sql.DB, queueName string) error {
+	// --- (1) ----
+	// Recover the indexes that the queue has
+	indexes, err := getIndexData(db, queueName)
+	if err != nil {
+		return err
+	}
+
+	// --- (2) ----
+	// Run through each one of the recovered indexes and see if the mandatory ones are included
+	var matchedIndexes int
+	for _, fields := range indexes {
+		if isIndexFound(fields) {
+			matchedIndexes++
+		}
+	}
+
+	// Check if we found all the mandatory indexes were found. If even 1 is missing, then we return an error
+	if matchedIndexes != len(mandatoryIndexes) {
+		return errors.Errorf("some PGQ indexes are missing")
+	}
+	return nil
+}
+
+func getColumnData(db *sql.DB, queueName string) (map[string]struct{}, error) {
+	rows, err := db.Query(columnSelect, queueName)
+	if err != nil {
+		return nil, errors.Wrap(err, "querying schema of queue table")
+	}
+	defer rows.Close()
+
+	columns := make(map[string]struct{})
+	for rows.Next() {
+		var s string
+		if err := rows.Scan(&s); err != nil {
+			return nil, errors.Wrap(err, "reading schema row of queue table")
+		}
+		columns[s] = struct{}{}
+	}
+	if err := rows.Err(); err != nil {
+		return nil, errors.Wrap(err, "reading schema of queue table")
+	}
+	if err := rows.Close(); err != nil {
+		return nil, errors.Wrap(err, "closing schema query of queue table")
+	}
+	if err != nil {
+		return nil, fmt.Errorf("error recovering queue columns %v", err)
+	}
+	return columns, nil
+}
+
+func getIndexData(db *sql.DB, queueName string) (map[string][]string, error) {
+	rows, err := db.Query(indexSelect, queueName)
+	if err != nil {
+		return nil, errors.Wrap(err, "querying index schema of queue table")
+	}
+	defer rows.Close()
+
+	indexes := make(map[string][]string)
+	for rows.Next() {
+		var indexName, indexColumns string
+		if err := rows.Scan(&indexName, &indexColumns); err != nil {
+			return nil, errors.Wrap(err, "reading index schema row of queue table")
+		}
+		indexes[indexName] = strings.Split(indexColumns, ",")
```

Comment:
Here is just `","`, no space
for `"foo, bar".split(",")` => `["foo", " bar"]` and `"bar" != " bar"`

Reply:
You are 100% correct... This was something I researched and the usage for it was to display the columns of the indexes rather than split them later on... I have made the changes, nice one 😄